src/db.rs

```diff
@@ -61,7 +78,7 @@ impl ReadonlyStorage for DB {
         start: Option<&[u8]>,
         end: Option<&[u8]>,
         order: cosmwasm_std::Order,
-    ) -> FfiResult<Box<dyn Iterator<Item = FfiResult<cosmwasm_std::KV>> + 'a>> {
+    ) -> FfiResult<Box<dyn Iterator<Item = StorageIteratorItem> + 'a>> {
```

Comment:
no gas counting here?

Just checked https://github.com/cosmos/cosmos-sdk/blob/v0.38.4/store/types/gas.go#L6-L15 and this is in fact unmetered. (Although I think that is a flaw.. it is as expensive as a query - can I just open iterators and never call next and it is "free"?).

I would pass out a gas used here (even if always 0) just for consistency. maybe i point this out the sdk devs and it changes

Reply:
I can change the `Box<dyn ...>` to `(Box<dyn ...>, u64)`, where the u64 will always be `0`, for now. Will you contact the cosmos devs about this?

Reply:
Note point below. They will actually charge for the first `Next` upon creating the iterator if it is valid. Querying only to find an empty range is however free - which could be abused by a malicious contract

Reply:
I will follow up with cosmos devs

Reply:
added gas counting for iterator creation in 622cb96